lib/ansible/modules/cloud/google/gce_labels.py

```diff
@@ -0,0 +1,315 @@
+short_description: Create, Update or Destory GCE Labels.
+description:
+    - Create, Update or Destory GCE Labels on instances, disks, snapshots, etc.
+      When specifying the GCE resource, users may specifiy the full URL for
+      the resource (its 'self_link'), or the individual parameters of the
+      resource (type, location, name). Examples for the two options can be
+      seen in the documentaion.
+      See U(https://cloud.google.com/compute/docs/label-or-tag-resources) for
+      more information about GCE Labels. Labels are gradually being added to
+      more GCE resources, so this module will need to be updated as new
+      resources are added to the GCE (v1) API.
+requirements:
+  - 'python >= 2.6'
+  - 'google-api-python-client >= 1.6.2'
+  - 'google-auth >= 1.0.0'
+  - 'google-auth-httplib2 >= 0.0.2'
+notes:
+  - Labels support resources such as  instances, disks, images, etc. See
+    U(https://cloud.google.com/compute/docs/labeling-resources) for the list
+    of resources available in the GCE v1 API (not alpha or beta).
+author:
+  - 'Eric Johnson (@erjohnso) <erjohnso@google.com>'
+options:
+  labels:
+    description:
+       - The list of labels (key/value pairs) to add to set on the resource.
+    required: false
+  resource_url:
+    description:
+       - The 'self_link' for the resource (instance, disk, snapshot, etc)
+    required: false
+  resource_type:
+    description:
+       - The type of resource (instances, disks, snapshots, images)
+    required: false
+  resource_location:
+    description:
+       - The location of resource (global, us-central1-f, etc.)
+    required: false
+  resource_name:
+    description:
+       - The name of resource.
+    required: false
+'''
+
+EXAMPLES = '''
+- name: Set GCE Labels on an existing instance (using resource_url)
+  gce_labels:
+    service_account_email: "{{ service_account_email }}"
+    credentials_file: "{{ credentials_file }}"
+    project_id: "{{ project_id }}"
+    labels:
+      webserver-frontend: homepage
+      environment: test
+      experiment-name: kennedy
+    resource_url: https://www.googleapis.com/compute/beta/projects/myproject/zones/us-central1-f/instances/example-instance
+    state: present
+- name: Set GCE Labels on an image (using resource params)
+  gce_labels:
+    service_account_email: "{{ service_account_email }}"
+    credentials_file: "{{ credentials_file }}"
+    project_id: "{{ project_id }}"
+    labels:
+      webserver-frontend: homepage
+      environment: test
+      experiment-name: kennedy
+    resource_type: images
+    resource_location: global
+    resource_name: my-custom-image
+    state: present
+- name: Clear all labels on an GCE image
+  gce_labels:
+    service_account_email: "{{ service_account_email }}"
+    credentials_file: "{{ credentials_file }}"
+    project_id: "{{ project_id }}"
+    resource_type: images
+    resource_location: global
+    resource_name: my-custom-image
+    state: absent
```

Comment:
@erjohnso I'm not familiar with the labels API, but if what @ryansb is asking for is true, there are some [helper functions in gce_tag](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/google/gce_tag.py#L133) that can handle the union, intersection, what's different, etc.